tardis/utilities/utils.py

```diff
@@ -66,6 +67,32 @@ def csv_parser(
             }
 
 
+KeyType = TypeVar("KeyType")
+
+
+def deep_update(
+    original_mapping: Mapping[KeyType, Any],
+    mapping_update: Mapping[KeyType, Any],
```

Comment:
I think technically the original must be `MutableMapping` since we need to change items in the copy. Consider to even restrict it to `Dict` since that's what is substituted at some points and you also special-case `List` (instead of `Sequence`).  
For `mapping_update` it's fine.

In the interest of generality, you *might* want to support `List` as well – that would allow a recursive `deep_update` into lists as well. Only if you want the extra work, though...